backend/functions/student/functions.js

```diff
@@ -37,17 +37,25 @@ const addStudentSurveyResponse = async (
     courseCatalogNames.map((name) => mapCatalogNameToCrseId(name))
   );
 
-  const studentUpdate = studentRef
+  const existingData = (await studentRef.doc(email).get()).data()
+  //studentCourses becomes courseIds of existingData.groups if available, otherwise []
+  let studentCrses = existingData ? existingData.groups : []
+  const studentCrseIds = studentCrses.map((crse)=> crse.courseId)
+  const crsesToAdd = crseIds.filter((crse) => (!studentCrseIds.includes(crse)))
+
+  crsesToAdd.map((crse) => studentCrses.push({
+    courseId: crse,
+    groupNumber:-1
+  }))
+
```

Comment:
Personally I'd do something like the following to combine the existing courses and new courses, but what you have is acceptable too (but change it to `forEach` instead of `map`, since not using the return value)

```js
const studentCourses = [...existingCourses, ...newCourses]
```